clcache.py

```diff
@@ -373,11 +373,14 @@ def save(self):
 
 
 class CacheStatistics:
-    def __init__(self, objectCache):
-        self.objectCache = objectCache
-        with objectCache.lock:
-            self._stats = PersistentJSONDict(os.path.join(objectCache.cacheDirectory(),
-                                                          "stats.txt"))
+    def __init__(self, cacheDirectory):
+        lockName = cacheDirectory.replace(':', '-').replace('\\', '-') + "_stats"
+        timeoutMs = 10 * 1000
+        self._lock = SystemSharedLock(lockName, timeoutMs)
+
+        with self._lock:
+            self._stats = PersistentJSONDict(os.path.join(cacheDirectory, "stats.txt"))
+
```

Comment:
Using two locks seems very dubious to me. Does this mean that the actual cache contents might now actually be different than what's printed in the statistics?